xbmc/interfaces/legacy/AddonUtils.h

```diff
@@ -68,6 +69,9 @@ namespace XBMCAddonUtils
   };
 
 #define LOCKGUI XBMCAddonUtils::GuiLock __gl
+#define LOCKGUIIF(cond) std::unique_ptr<XBMCAddonUtils::GuiLock> __gl; \
```

Comment:
another problem is that LOCKGUI on list items seems to be broken in general because there is no language hook in thread local storage.